source/_UIAHandler.py

```diff
@@ -182,11 +182,16 @@ def MTAThreadFunc(self):
 			except (COMError,WindowsError,NameError):
 				self.clientObject=CoCreateInstance(CUIAutomation._reg_clsid_,interface=IUIAutomation,clsctx=CLSCTX_INPROC_SERVER)
 			if isUIA8:
-				try:
-					self.clientObject=self.clientObject.QueryInterface(IUIAutomation3)
-				except COMError:
-					self.clientObject=self.clientObject.QueryInterface(IUIAutomation2)
-			log.info("UIAutomation: %s"%self.clientObject.__class__.__mro__[1].__name__)
+				# #8009: use appropriate interface based on highest supported interface.
+				for interface in (IUIAutomation5, IUIAutomation4, IUIAutomation3, IUIAutomation2):
+					try:
+						self.clientObject=self.clientObject.QueryInterface(interface)
+						break
+					except COMError:
+						pass
+			# Have this handy because we want to add things corresponding to the interface version.
+			IUIAVersion = self.clientObject.__class__.__mro__[1].__name__

```

Comment:
Now you loop through these interfaces (which I think is a good thing to do), you might as well do something like this, avoiding digging into the mro:

```
for interface in interfaces:
    QueryInterface
except:
    pass # Or continue? I guess they are identical in result here
else:
    IUIAVersion = interface.__name__
    ...
```

Reply:
UiaVersion would also have to be set to IUIAutomation when isUIA8 is false. if you do it the way @leonardder suggests